google/cloud/aiplatform/training_jobs.py

```diff
@@ -4503,7 +4516,9 @@ def _run(
             "quantiles": quantiles,
             "validationOptions": validation_options,
             "optimizationObjective": self._optimization_objective,
+            "holidayRegions": holiday_regions,
         }
+        self._validate_training_task_inputs(training_task_inputs_dict)
```

Comment:
We typically leave validation to the service, so we don't have to maintain two sources of truth. Do you see an issue with this?

It may add a bit more wait time for the user but it vastly reduces complexity.

Reply:
I've seen validation for things like the splitting strategy: https://github.com/googleapis/python-aiplatform/blob/main/google/cloud/aiplatform/training_jobs.py#L521

I think it would be a better user experience to do some simple validation where we can, can't see how it would hurt. The wait time is fairly significant since input validation occurs in the training pipeline, not in the request handler, meaning we need to wait for resources to spin up and there is a resource cost associated with passing in poor inputs via the SDK.

Reply:
Also, do you know if there is a way to validate inputs against a given [training task definition](https://source.corp.google.com/piper///depot/google3/google/cloud/aiplatform/publicfiles/trainingjob/definition/automl_time_series_forecasting_1.0.0.yaml) on the SDK side? Seems like a fairly simple and reasonable thing to do.